src/witan/cic/tuairisc/summary/census.clj

```diff
@@ -0,0 +1,61 @@
+(ns witan.cic.tuairisc.summary.census
+  (:require
+   [ham-fisted.api :as hf]
+   [ham-fisted.reduce :as hfr]
+   [tablecloth.api :as tc]
+   [tick.core :as tick]
+   [tech.v3.dataset :as ds]
+   [tech.v3.dataset.reductions :as dsr]
+   [witan.cic.tuairisc.date-utils :as du]
+   [witan.cic.tuairisc.summary :as summary]))
+
+;; ds/row-mapcat is too slow
+(defn range->census [projection-episodes {:keys [analysis-range max-batch-size]
+                                          :or {max-batch-size 10}}]
+  (let [analysis-start (reduce tick/min analysis-range)
+        analysis-end (reduce tick/max analysis-range)]
+    (as-> projection-episodes $
+      (tc/select-rows $ #(du/overlaps-window? analysis-start analysis-end (% :episode-start) (% :episode-end)))
+      (ds/row-mapcat $ (summary/census-extractor-fn analysis-range) {:max-batch-size max-batch-size :result-type :as-seq}))))
+
+;; I think this version should be faster than the ds/row-mapcat
+;; version, but it seems to fail after running a few times and I
+;; haven’t been able to figure out why. I’m leaving it here as I’d
+;; like to move over to this code eventually
+#_
+(defn range->census [projection-episodes {:keys [analysis-range max-batch-size]
+                                          :or {max-batch-size 10}}]
+  (let [analysis-start (reduce tick/min analysis-range)
+        analysis-end (reduce tick/max analysis-range)]
+    (as-> projection-episodes $
+      (tc/select-rows $ #(du/overlaps-window? analysis-start analysis-end (% :episode-start) (% :episode-end)))
+      (tc/rows $ :as-maps)
+      (hfr/preduce
+       ;; init-val
+       (fn [] (hf/mut-list))
+       ;; rfn
+       (fn [acc x] (let [ds (tc/dataset ((summary/census-extractor-fn analysis-range) x))]
+                     (if (zero? (tc/row-count ds))
+                       acc
+                       (hf/conj! acc ds))))
+       ;; merge-fn
+       (fn [x x'] (hf/add-all! x x'))
+       $)
+      (hf/immut-list $))))
```

Comment:
I think you need to comment this out correctly otherwise when you call this library this is the version of the fn that will be used

Reply:
or change the name of this fn (v2?)

Reply:
the api is the same, so this should be a direct replacement and not need a v2. The #_ comments out the next sexp so works even though it is on the line above.

Reply:
there wasn't a summary namespace like this before.

I'm not sure some of the older summary code was ever used successfully anywhere (for costs).